pkg/asset/installconfig/vsphere/metadata.go

```diff
@@ -122,10 +125,32 @@ func (m *Metadata) unlockedSession(ctx context.Context, server string) (*session
 	return m.sessions[server], err
 }
 
+func unwrapToSoapFault(err error) error {
+	if err != nil {
+		soapFault := soap.IsSoapFault(err)
+		if !soapFault {
+			return unwrapToSoapFault(errors.Unwrap(err))
+		}
+		return err
+	}
+	return err
+}
+
```

Comment:
So you cannot use `errors.As` to do this for you?

Reply:
soapFaultError isn't exported, kinda annoying ;-)

https://github.com/vmware/govmomi/blob/c385285e275e4bb07a92aa46b59c19f98db2fd5d/vim25/soap/error.go#L38